static/js/compose.js

```diff
@@ -1037,13 +1039,38 @@ exports.initialize = function () {
         if (page_params.realm_video_chat_provider === available_providers.google_hangouts.id) {
             video_call_link = "https://hangouts.google.com/hangouts/_/" + page_params.realm_google_hangouts_domain + "/" + video_call_id;
             insert_video_call_url(video_call_link, target_textarea);
-        } else if (page_params.realm_video_chat_provider === available_providers.zoom.id) {
-            channel.get({
-                url: '/json/calls/create',
-                success: function (response) {
-                    insert_video_call_url(response.zoom_url, target_textarea);
-                },
-            });
+        } else if (available_providers.zoom
+            && page_params.realm_video_chat_provider === available_providers.zoom.id) {
+            let zoom_window = null;
+            const make_zoom_call = () => {
+                exports.got_zoom_token = () => {};
+                if (zoom_window) {
+                    zoom_window.close();
+                }
+                channel.post({
+                    url: "/json/make_zoom_video_call",
+                    success(res) {
+                        insert_video_call_url(res.url, target_textarea);
+                    },
+                    error(xhr, error_type, xhn) {
+                        window.alert("error!");
```

Comment:
This error handler should use `compose_error` to put an error message above the compose box containing a link to retry.  And the server should send an event notice that the token is no longer valid.

Reply:
Obviously that was a placeholder error handler, but I have now discovered that we might not even be in the compose box—we might be editing a previously sent message.

Reply:
Hmm, yeah; I think we don't have a convenient way to place errors for that use case, since it didn't previously have this category of error.  I'm fine with just having the error be ~silent there in this commit and then opening an issue for doing better; it's not worth adding another moving part here.